### ISSUE TITLE
Rule Change: Add suggestion to remove unused member in `no-unused-private-class-members`

### ISSUE THREAD

**Issue by cocomittens** [eslint/eslint#20516]
### What rule do you want to change?

no-unused-private-class-members

### What change do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
class Foo {
  #unusedField = 1;
  #unusedMethod() {}
}
```

### What does the rule currently do for this code?

The rule reports both the unused field and method as errors, but provides no fix or suggestion.

### What will the rule do after it's changed?

The rule will report both the unused field and method as errors, as well as provide a suggestion to remove the unused member's definition from the class. 

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by mdjermanovic** [comment]
Makes sense to me 👍 

@eslint/eslint-team thoughts?

**Comment by lumirlumir** [comment]
Personally, I'm also +1 on implementing the suggestion for this rule. 👍

**Comment by DMartens** [comment]
I am also in favor of this, so marking as accepted.
Unfortunately you cannot simply remove the unused class member as you have to make sure of missing semicolons.
For example removing `#unused` in the following example leads to a parse error:
```javascript
class Class {
	before = fn()
	#unused;
	[computed]() {}
}
```
The reason is that the following computed class member is then interpreted as `fn()[computed](){}` and not as a separate class method, so you have to add a semicolon after `fn()`.
Feel free not to handle this in the initial PR and to reach out if you need help implementing this (e.g. via `#contributors` channel in our discord).

**Comment by sethamus** [comment]
@cocomittens are you planning to submit a PR for this? Otherwise I can put one up.

**Comment by cocomittens** [comment]
@sethamus Yes I am planning to submit a PR soon

**Comment by mdjermanovic** [comment]
Assigning to @sethamus as per https://github.com/eslint/eslint/pull/20557#issuecomment-4174011364
